dashboard-widgets.el

```diff
@@ -225,128 +225,100 @@ If MESSAGEBUF is not nil then MSG is also written in message buffer."
           (dashboard-insert-ascii-banner-centered banner))))))
 
 ;;
-;; Recentf
+;; Section insertion
 ;;
-(defun dashboard-insert-recentf-list (list-display-name list)
-  "Render LIST-DISPLAY-NAME title and items of LIST."
-  (when (car list)
-    (dashboard-insert-heading list-display-name)
-    (mapc (lambda (el)
-            (insert "\n    ")
-            (widget-create 'push-button
-                           :action `(lambda (&rest ignore) (find-file-existing ,el))
-                           :mouse-face 'highlight
-                           :follow-link "\C-m"
-                           :button-prefix ""
-                           :button-suffix ""
-                           :format "%[%t%]"
-                           (abbreviate-file-name el)))
-          list)))
+(defmacro dashboard-insert-section-list (section-name list action &rest rest)
+	"Insert a LIST of items with SECTION-NAME, expanding ACTION and passing REST to widget creation."
+	`(let ((max-line-length 0))
+		 (when (car ,list)
+			 (mapc (lambda (el)
+							 (let ((widget nil))
+								 (insert "\n    ")
+								 (setq widget
+											 (widget-create 'push-button
+																			:action ,action
+																			:mouse-face 'highlight
+																			:follow-link "\C-m"
+																			:button-prefix ""
+																			:button-suffix ""
+																			:format "%[%t%]"
+																			,@rest))
+								 (setq max-line-length
+											 (max max-line-length (length (widget-value-value-get widget))))))
+						 ,list))
+		 max-line-length))
+
+(defmacro dashboard-insert-section (section-name list list-size shortcut action &rest widget-params)
+  "Add a section with SECTION-NAME and LIST of LIST-SIZE items to the dashboard.
+ACTION is theaction taken when the user activates the widget button.
+SHORTCUT is the keyboard shortcut used to access the section.
+WIDGET-PARAMS are passed to the \"widget-create\" function.
+Show EMPTY-LIST-TEXT if no items in list"
+	`(progn
+		 (dashboard-insert-heading ,section-name)
+		 (when-let ((max-line-length
+								 (dashboard-insert-section-list
```

Comment:
Indentation is wrong here and the lines below

Reply:
Fixed!